network/server/tchannelthrift/node/service.go

```diff
@@ -277,3 +278,8 @@ func (s *service) WriteBatch(tctx thrift.Context, req *rpc.WriteBatchRequest) er
 	}
 	return nil
 }
+
+func (s *service) TruncateNamespace(tctx thrift.Context, req *rpc.TruncateNamespaceRequest) error {
+	s.db.TruncateNamespace(req.Ns)
+	return nil
```

Comment:
Yeah can do, just saying we don't need to deal with existing shards with the current implementation, but sure